.travis.yml

```diff
@@ -1,188 +1,109 @@
-# and install a latest Java 11. For reference:
-#   https://sormuras.github.io/blog/2017-12-08-install-jdk-on-travis.html
-# Java12 does not have this issue, we do not download the Java11 script
-# for that JVM or else we would be overriding Java12 with Java11.  
-#
-# Additional note: The export commands seem to not apply when extracted
-# to a script and seemingly only work when located in the main travis config
-# here.
-#
+jdk: openjdk11
+cache: { directories: [ "$HOME/.gradle" , ".gradle" ] }
+install: skip
 
+env:
+  global:
+   - REPORT_FAIL=".travis/shared/report-build-status FAILURE"
+   - VERIFY_STAGE=".travis/stages/verify"
+   - DEPLOY_STAGE=".travis/stages/deploy"
 jobs:
   include:
-    - stage: verify
-      env: "Static Verification"
-      jdk: openjdk11
-      cache:
-        directories:
-          - '$HOME/.gradle'
-          - '.gradle'
-          - '~/jdk-11.0.5+10'
-      before_install:
-        - ./.travis/install_java_11.0.5
-        - export JAVA_HOME=~/jdk-11.0.5+10
-        - export PATH=${JAVA_HOME}/bin:$PATH
-        - java --version
-      install: skip
-      script:
-        ## shellcheck all shell scripts
-        - find .travis/ -type f | xargs grep -lE "^#\!/bin/bash$" | xargs shellcheck
-        - .travis/check-links || travis_terminate 1;
-        - .travis/find-unused-dbunit-datasets || travis_terminate 1
-        - .travis/find-unmatched-test-packages || travis_terminate 1;
-        - .travis/find-orphaned-tests || travis_terminate 1;
-        - .travis/check-custom-style || travis_terminate 1;
-        - ./gradlew --parallel validateYamls spotlessCheck checkstyleMain checkstyleTest pmdMain pmdTest
-    - stage: verify
-      env: "Unit & Integration Tests"
-      jdk: openjdk11
-      cache:
-        directories:
-          - '$HOME/.gradle'
-          - '.gradle'
-          - '~/jdk-11.0.5+10'
-      addons: {postgresql: "10"}
-      before_install:
-        - ./.travis/install_java_11.0.5
-        - export JAVA_HOME=~/jdk-11.0.5+10
-        - export PATH=${JAVA_HOME}/bin:$PATH
-        - java --version
-      install: skip
-      script:
-        - .travis/setup_database
-        - .travis/launch-http-server || travis_terminate 1
-        - ./gradlew --parallel test integTest jacocoTestReport || travis_terminate 1
-        - bash <(curl -s https://codecov.io/bash)  # upload coverage report - https://github.com/codecov/example-gradle
-    - stage: verify
-      env: "Smoke Test"
-      jdk: openjdk11
-      cache:
-        directories:
-          - '$HOME/.gradle'
-          - '.gradle'
-          - '~/jdk-11.0.5+10'
+    
+    ########################################################################
+    # Jobs are ordered tests first and then slower jobs before faster ones.
+  
+    - stage: Verify
+      name: "Tests: Integration Test"
+      addons: { postgresql: "10" }
+      script: [ $VERIFY_STAGE/integration-testing/run || $REPORT_FAIL ]
```

Comment:
For all of the script parts: I'm pretty sure that travis doesn't care if a list or a single entry is given. After all the default value for this key should be `./gradlew check` IIRC

Reply:
Hmm, not sure if this matters. If we wind up adding more, then it'll be less of a diff, on the other hand we should not be adding more lines as each item in 'script' is intentionally extracted to 'run'.

Reply:
Took a second look at this, while the list syntax is not strictly necessary, it does do a good job to visually distinguish the script step from the other labels that are scalars.